src/allotropy/parsers/novabio_flex2/novabio_flex2_parser.py

```
@@ -27,9 +28,13 @@ class NovaBioFlexParser(VendorParser[Data, Model]):
 
     def create_data(self, named_file_contents: NamedFileContents) -> Data:
         # NOTE: calling parse_dates and removing NaN clears empty rows.
-        data = read_csv(
-            named_file_contents.contents, parse_dates=["Date & Time"]
-        ).replace(np.nan, None)
+        try:
+            data = read_csv(
+                named_file_contents.contents, parse_dates=["Date & Time"]
+            ).replace(np.nan, None)
+        except Exception as e:
+            msg = "NovaBio Flex2 parser requires 'Date & Time' column to be present in the CSV file."
```

We can't catch all exceptions from this call and return this error message.

If the call fails for another reason, the message will be misleading.

Instead, we should check for the "Date & Time" column first, and throw an error if missing.

---

Ah right, we don't have the pd columns yet.

What you could do is:

1. Only except ValueError
2. Check that the error message includes "Missing column provided to 'parse_dates'"